File: Java/HNO/src/main/java/com/app/service/DonorService.java
package com.app.service;

import java.util.List;

import com.app.dto.Donor.DonorLoginDto;
import com.app.pojos.BankTransaction;
import com.app.pojos.Donor;

public interface DonorService {
	//Crud Operations
	List<Donor> getAllDonors(); 
	Donor addDonor(Donor donor);
	String deleteDonor(Long donorId);
	Donor getOneDonor(Long donId);
	Donor updateDonor(Donor don);
	//Main Functins
	Donor validateDon(DonorLoginDto don);
	List<Donor> getAllTrueDonors();
	String deleteFalseDonor(Long donorId);
	BankTransaction depositBankTransaction(BankTransaction transaction);
//	BankTransaction withdrawBankTransaction(BankTransaction Transaction);
}

File: Front End/Practice/Local storage vs Session storage/form_with_session/src/Component/Services/DonorService.js
import axios from "axios";

class Service{
    constructor(){
        this.url="http://localhost:8080/donors";
    }
    GetAllDonors(){
        return axios.get(this.url+"/Donortrue");
    }
    AddDonor(donor){
        return axios.post(this.url,donor);
    }
    SignInDonor(donor){
        return axios.post(this.url+"/signIn",donor);
    }
    DeleteDonor(id){
        return axios.put(this.url+"/",id);
    }
    UpdateDonor(donor){
        return axios.put(this.url,donor);
    }
}
export default new Service();